OlinkAnalyze/R/olink_normalization.R

```diff
@@ -596,35 +628,71 @@ norm_internal_cross_product <- function(ref_df,
     not_ref_df = not_ref_df,
     not_ref_name = not_ref_name,
     not_ref_cols = not_ref_cols
+  ) |>
+    # keep only relevant columns
+    dplyr::select(
+      dplyr::all_of(
+        c(ref_cols$sample_id, ref_cols$olink_id, "Project",
+          "MedianCenteredNPX" = ref_cols$quant)
+      )
+    )
+
+  # quantile normalize HT-3k ----
+
+  df_norm_qq <- olink_normalization_qs(
+    lst_df = lst_df,
+    ref_cols = ref_cols,
+    bridge_samples = ref_samples
   )
 
-  # when ref and non-ref datasets are merged during bridging, assay identifiers
-  # OlinkID_HT and OlinkID_E3072 are NA. Here we fill them in.
-  df_norm_bridge <- df_norm_bridge |>
+  # integrate normalization approaches ----
+
+  # combines df_is_bridgeable, df_norm_bridge and df_norm_qq
+  # stores the outcome in df_norm
+  df_norm <- lst_df |>
+    dplyr::bind_rows(
+      .id = "Project"
+    ) |>
+    # when ref and non-ref datasets are merged during bridging, assay
+    # identifiers OlinkID_HT and OlinkID_E3072 are NA. Here we fill them in.
     dplyr::group_by(
       .data[[ref_cols$olink_id]]
     ) |>
     tidyr::fill(
       dplyr::starts_with(ref_cols$olink_id),
       .direction = "updown"
     ) |>
-    dplyr::ungroup()
-
-  # quantile normalize HT-3k ----
-
-  # to be filled in
-  # returns df_norm_qq
-
-  # integrate normalization approaches ----
-
-  # to be filled in
-  # combines df_is_bridgeable, df_norm_bridge and df_norm_qq
-  # stores the outcome in df_norm
-  df_norm <- df_norm_bridge |>
+    dplyr::ungroup() |>
+    # add bridge normalized values
+    dplyr::left_join(
+      df_norm_bridge,
+      by = c(ref_cols$sample_id, ref_cols$olink_id, "Project"),
+      relationship = "one-to-one"
+    ) |>
+    # add bridgeable
```

Comment:
Can we move this part after df_norm_qq, so the order of the columns be "MedianCenteredNPX"   "QSNormalizedNPX"   "BridgingRecommendation"?

Reply:
I guess you mean joining the results with the following order:
1. add bridge normalized
2. add QS normalized
3. add is bridgeable.
Fixed.